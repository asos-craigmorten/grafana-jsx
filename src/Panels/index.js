import PropTypes from "prop-types";

const Panels = ({ children }) => ({
  panels: children,
});

Panels.PropTypes = {
  children: PropTypes.arrayOf(PropTypes.object),
};

export default Panels;
