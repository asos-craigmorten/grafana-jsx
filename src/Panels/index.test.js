import Panels from "./";

describe("Panels", () => {
  describe("when passed no Panel children", () => {
    it("should render", () => {
      expect(<Panels />).toMatchSnapshot();
    });
  });

  describe("when passed Panel children", () => {
    it("should render", () => {
      expect(
        <Panels>
          <test-panel test-panel-prop={Symbol("test-panel-value")} />
          <test-panel test-panel-prop={Symbol("test-panel-value")} />
        </Panels>
      ).toMatchSnapshot();
    });
  });
});
