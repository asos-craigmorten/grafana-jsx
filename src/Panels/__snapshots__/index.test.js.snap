// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panels when passed Panel children should render 1`] = `
Object {
  "panels": Array [
    Object {
      "test-panel-prop": Symbol(test-panel-value),
    },
    Object {
      "test-panel-prop": Symbol(test-panel-value),
    },
  ],
}
`;

exports[`Panels when passed no Panel children should render 1`] = `
Object {
  "panels": Array [],
}
`;
